src/app/app.module.ts

```diff
@@ -7,11 +7,20 @@ import { AppComponent } from './app.component';
 import { CoursesModule } from './features/courses/courses.module';
 import { LoginModule } from './features/login/login.module';
 import { RegistrationModule } from './features/registration/registration.module';
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { TokenInterceptor } from './auth/interceptors/token.interceptor';
+import { AuthModule } from './auth/auth.module';
 
 @NgModule({
   declarations: [AppComponent],
-  imports: [BrowserModule, AppRoutingModule, CoursesModule, LoginModule, RegistrationModule],
-  providers: [],
+  imports: [BrowserModule, AppRoutingModule, CoursesModule, LoginModule, RegistrationModule, HttpClientModule, AuthModule],
```

Comment:
If I remember correctly, you can remove imports here for modules which is using for lazy loading.
`CoursesModule, LoginModule, RegistrationModule`.

Reply:
removed